engine/src/flutter/lib/web_ui/lib/src/engine/renderer.dart

```diff
@@ -226,10 +263,78 @@ abstract class Renderer {
 
   ui.ParagraphBuilder createParagraphBuilder(ui.ParagraphStyle style);
 
-  Future<void> renderScene(ui.Scene scene, EngineFlutterView view);
+  /// Map from view id to the associated [ViewRasterizer] for that view.
+  final Map<int, ViewRasterizer> rasterizers = <int, ViewRasterizer>{};
+
+  Future<void> renderScene(ui.Scene scene, EngineFlutterView view) async {
+    assert(
+      rasterizers.containsKey(view.viewId),
+      "Unable to render to a view which hasn't been registered",
+    );
+    final ViewRasterizer rasterizer = rasterizers[view.viewId]!;
+    final RenderQueue renderQueue = rasterizer.queue;
+    final FrameTimingRecorder? recorder = FrameTimingRecorder.frameTimingsEnabled
+        ? FrameTimingRecorder()
+        : null;
+    if (renderQueue.current != null) {
+      // If a scene is already queued up, drop it and queue this one up instead
+      // so that the scene view always displays the most recently requested scene.
+      renderQueue.next?.completer.complete();
+      final Completer<void> completer = Completer<void>();
+      renderQueue.next = (scene: scene, completer: completer, recorder: recorder);
+      return completer.future;
+    }
+    final Completer<void> completer = Completer<void>();
+    renderQueue.current = (scene: scene, completer: completer, recorder: recorder);
+    unawaited(_kickRenderLoop(rasterizer));
+    return completer.future;
+  }
+
+  Future<void> _kickRenderLoop(ViewRasterizer rasterizer) async {
+    final RenderQueue renderQueue = rasterizer.queue;
+    final RenderRequest current = renderQueue.current!;
+    try {
+      await _renderScene(current.scene, rasterizer, current.recorder);
+      current.completer.complete();
+    } catch (error, stackTrace) {
+      current.completer.completeError(error, stackTrace);
+    }
+    renderQueue.current = renderQueue.next;
+    renderQueue.next = null;
+    if (renderQueue.current == null) {
+      return;
+    } else {
+      return _kickRenderLoop(rasterizer);
+    }
+  }
+
+  Future<void> _renderScene(
+    ui.Scene scene,
+    ViewRasterizer rasterizer,
+    FrameTimingRecorder? recorder,
+  ) async {
+    await rasterizer.draw((scene as LayerScene).layerTree, recorder);
+    recorder?.submitTimings();
+  }
 
   void dumpDebugInfo();
 
+  /// Disposes this renderer.
+  @mustCallSuper
+  void dispose() {
+    _onViewCreatedListener.cancel();
+    _onViewDisposedListener.cancel();
+    for (final ViewRasterizer rasterizer in rasterizers.values) {
+      rasterizer.dispose();
+    }
+    rasterizers.clear();
```

Comment:
Should we also dispose of the `Rasterizer`?
```suggestion
    for (final ViewRasterizer rasterizer in rasterizers.values) {
      rasterizer.dispose();
    }
    rasterizers.clear();
    rasterizer.dispose();
```